src/util/glTF/glTFContainer.js

```diff
@@ -127,43 +128,91 @@ x3dom.glTF.glTFTexture.prototype.isPowerOfTwo = function(x)
     return powerOfTwo;
 };
 
+x3dom.glTF.glTFTexture.prototype.needsPowerOfTwo = function(gl)
+{
+    var resize = true;
+    resize &= (this.sampler.magFilter == gl.LINEAR || this.sampler.magFilter == gl.NEAREST);
+    resize &= (this.sampler.minFilter == gl.LINEAR || this.sampler.minFilter == gl.NEAREST);
+    resize &= (this.sampler.wrapS == gl.CLAMP_TO_EDGE);
+    resize &= (this.sampler.wrapT == gl.CLAMP_TO_EDGE);
+
+    return !resize;
+};
+
+x3dom.glTF.glTFTexture.prototype.needsMipMaps = function(gl)
+{
+    var need = true;
+    need &= (this.sampler.magFilter == gl.LINEAR || this.sampler.magFilter == gl.NEAREST);
+    need &= (this.sampler.minFilter == gl.LINEAR || this.sampler.minFilter == gl.NEAREST);
+
+    return !need;
+};
+
+x3dom.glTF.glTFTexture.prototype.nextPowerOfTwo = function(n)
```

Comment:
We have already a utils function to get the next power of two, maybe it make more sense to use that.
https://github.com/x3dom/x3dom/blob/master/src/util/Utils.js#L675